streams/src/main/java/org/apache/kafka/streams/state/internals/RocksDBStore.java

```diff
@@ -172,6 +169,11 @@ public void openDB(ProcessorContext context) {
             valueSerde == null ? (Serde<V>) context.valueSerde() : valueSerde);
 
         this.dbDir = new File(new File(context.stateDir(), parentDir), this.name);
+
+        if (!hasSstFiles(dbDir) && prepareForBulkload) {
+            options.prepareForBulkLoad();
```

Comment:
If we decide to not turn on bulk loading, then we should be able to not close and restart again after restoring is done.